internal/config/storage.go

```diff
@@ -13,6 +13,7 @@ const (
 	DatabaseStorageType = StorageType("database")
 	LocalStorageType    = StorageType("local")
 	GitStorageType      = StorageType("git")
+	S3StorageType       = StorageType("s3")
```

Comment:
wdyt about generalizing the config for object/blob storage to make it more extensible in the future (`gcp`, `azure`, etc)?

Like for example if the config looked something like this:

```yaml
storage:
  object: # or blob
    s3:
```

Nesting it under `blob` or `object` would allow for extensibility without adding a new 'top-level' key to support these alternative stores.

Right now we would only support `storage.object.s3`, but in the future if other providers were added we would make them mutually exclusive and error if you provided configuration options more more than one

Reply:
Ok, I can see the idea of trying to plan a bit for the future. I'll update the PR to reflect that idea

Reply:
@markphelps PR updated. Let me know if it is in line with what you had in mind.

Thank you for the suggestion